packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -438,32 +438,39 @@ export default class SegmentationStateManager {
 
   /**
    * Helper function to update labelmap segmentation image references.
-   * @param {string} segmentationId - The ID of the segmentation representation.
-   * @param {Types.IViewport} viewport - The viewport.
-   * @param {string[]} labelmapImageIds - The labelmap image IDs.
-   * @param {Function} updateCallback - A callback to update the reference map.
+   * @param segmentationInfo - The segmentation info of the desired segment.
+   * @param viewport - The viewport.
+   * @param labelmapImageIds - The labelmap image IDs.
+   * @param updateCallback - A callback to update the reference map.
    * @returns {string | undefined} The labelmap imageId reference for the current imageId rendered on the viewport.
    */
   _updateLabelmapSegmentationReferences(
-    segmentationId,
-    viewport,
-    labelmapImageIds,
-    updateCallback
+    segmentationInfo: Segmentation,
+    viewport: Types.IStackViewport,
+    labelmapImageIds: string[],
+    updateCallback: (
+      viewport: Types.IStackViewport,
+      segmentationId: string,
+      labelmapImageIds: string[]
+    ) => void
   ): string | undefined {
     const currentImageId = viewport.getCurrentImageId();
+    const currentImageIndex = viewport.getCurrentImageIdIndex();
+    const segmentationId = segmentationInfo.segmentationId;
 
     let viewableLabelmapImageIdFound = false;
-    for (const labelmapImageId of labelmapImageIds) {
+    const labelForImageIndex = labelmapImageIds[currentImageIndex];
+    if (labelForImageIndex) {
```

Comment:
i always wondered why we don't do this instead of looping over the labelmapImageIds and finding the match.